feat(design): implementation for style-dictionary (#2807)
AmadeusITGroup/otter: packages/@o3r/style-dictionary/README.md

| o3rScope          | `string`                                                                                                                                        | Scope to apply to the generated variable                                                                                                                                                                                                                                            | pre-processor: **o3r/pre-processor/extensions** <br />formatter: **o3r/css/variable**     |
| o3rMetadata       | [CMS Metadata](https://github.com/AmadeusITGroup/otter/blob/main/packages/%40o3r/style-dictionary/src/interfaces/stye-dictionary.interface.mts) | Additional information to provide to the metadata if generated                                                                                                                                                                                                                      | pre-processor: **o3r/pre-processor/extensions** <br />formatter: **o3r/json/metadata**    |

> | o3rUnit           | `string`                                                                                                                                        | Convert a numeric value from the specified unit to the new unit. It will add a unit to the token with type \"number\" for which the unit is not specified.<br />In case of complex type (such as shadow, transition, etc...), the unit will be applied to all numeric types in it   | pre-processor: **o3r/pre-processor/extensions** <br />transforms: **o3r/transform/unit**  |

Reviewer: ```suggestion
| o3rUnit           | `string`                                                                                                                                        | Convert a numeric value from the specified unit to the new unit. It will add a unit to the tokens of type \"number\" for which the unit is not specified.<br />In the case of complex types (such as shadow, transition, etc...), the unit will be applied to all numeric types they contain.   | pre-processor: **o3r/pre-processor/extensions** <br />transforms: **o3r/transform/unit**  |
```

> | o3rUnit                | `string`                                                                                                                                        | Convert a numeric value from the specified unit to the new unit. It will add a unit to the token with type \"number\" for which the unit is not specified.\nIn case of complex type (such as shadow, transition, etc...), the unit will be applied to all numeric types in it   | pre-processor: **o3r/pre-processor/extensions** <br />transforms: **o3r/transform/unit**  |

Reviewer 2: ```suggestion
| o3rUnit                | `string`                                                                                                                                        | Convert a numeric value from the specified unit to the new unit. It will add a unit to the token with type \"number\" for which the unit is not specified.<br />In case of complex type (such as shadow, transition, etc...), the unit will be applied to all numeric types in it   | pre-processor: **o3r/pre-processor/extensions** <br />transforms: **o3r/transform/unit**  |
```

> | o3rRatio          | `number`                                                                                                                                        | Ratio to apply to previous value. The ratio will be applied only on token with \"number\" type or on the first numbers determined in \"string\" like types.<br />In case of complex type (such as shadow, transition, etc...), the ratio will be applied to all numeric types in it | pre-processor: **o3r/pre-processor/extensions** <br />transforms: **o3r/transform/ratio** |

Reviewer: ```suggestion
| o3rRatio          | `number`                                                                                                                                        | Ratio to apply to the previous value. The ratio will only be applied to tokens of type \"number\" or to the first numbers determined in \"string\" like types.<br />In the case of complex types (such as shadow, transition, etc...), the ratio will be applied to all numeric types they contain. | pre-processor: **o3r/pre-processor/extensions** <br />transforms: **o3r/transform/ratio** |
```

> | o3rRatio               | `number`                                                                                                                                        | Ratio to apply to previous value. The ratio will be applied only on token with \"number\" type or on the first numbers determined in \"string\" like types.\nIn case of complex type (such as shadow, transition, etc...), the ratio will be applied to all numeric types in it | pre-processor: **o3r/pre-processor/extensions** <br />transforms: **o3r/transform/ratio** |

Reviewer 2: Are we s
```suggestion
| o3rRatio               | `number`                                                                                                                                        | Ratio to apply to previous value. The ratio will be applied only on token with \"number\" type or on the first numbers determined in \"string\" like types.<br />In case of complex type (such as shadow, transition, etc...), the ratio will be applied to all numeric types in it | pre-processor: **o3r/pre-processor/extensions** <br />transforms: **o3r/transform/ratio** |
```

| o3rExpectOverride | `boolean`                                                                                                                                       | Indicate that the token is expected to be overridden by external rules                                                                                                                                                                                                              | pre-processor: **o3r/pre-processor/extensions**                                           |

> [!WARNING]

> > The **Required Hooks** need to be registered to the [Style Dictionary Config](https://styledictionary.com/reference/config/) to fully support the extension.

Reviewer: ```suggestion
> The **required hooks** need to be registered to the [Style Dictionary configuration](https://styledictionary.com/reference/config/) to fully support the extension.
```

### [Pre-processor](https://styledictionary.com/reference/hooks/preprocessors/)

| Name                         | Description                                                                                                                                                                 |
| ---------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| o3r/pre-processor/extensions | Pre-processor to add the support of the `$extensions` instructions in the Token (or dedicated `extensions.json` file). This pre-processor is mandatory for any Otter hooks. |

### [Transforms](https://styledictionary.com/reference/hooks/transforms/)

| Name                | Type    | Criteria                          | Description                                                                                  |
| ------------------- | ------- | --------------------------------- | -------------------------------------------------------------------------------------------- |

> | o3r/transform/ratio | `value` | extensions `o3rRatio` is provided | Apply the given `o3rRatio` to the numeric values of the Token(s) it refer to.                |

Reviewer: ```suggestion
| o3r/transform/ratio | `value` | `o3rRatio` extension is provided | Apply the given `o3rRatio` to the numeric values of the Token(s) it refers to.                |
```

> | o3r/transform/unit  | `value` | extensions `o3rUnit` is provided  | Replace the unit of the values of the Token(s) it refer to, by the provided given `o3rUnit`. |

Reviewer: ```suggestion
| o3r/transform/unit  | `value` | `o3rUnit` extension is provided  | Replace the unit of the values of the Token(s) it refers to, by the provided `o3rUnit`. |
```

| o3r/css/recommended | Extend the official [CSS transform groups](https://styledictionary.com/reference/hooks/transform-groups/predefined/#css) by adding `o3r/transform/ratio` and `o3r/transform/unit` |

### [Formats](https://styledictionary.com/reference/hooks/formats/)

| Name              | Options                                                                                                                                 | Description                                                                                                                                                                                             |
| ----------------- | --------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | o3r/css/variable  | See [css/variables options](https://styledictionary.com/reference/hooks/formats/predefined/#cssvariables)                               | Renderer CSS Variable block (based on [built-in format](https://styledictionary.com/reference/hooks/formats/predefined/#cssvariables)) supporting additional [Otter Extensions features](#enhancement). |

Reviewer: ```suggestion
| o3r/css/variable  | See [css/variables options](https://styledictionary.com/reference/hooks/formats/predefined/#cssvariables)                               | Render CSS variable block (based on the [built-in format](https://styledictionary.com/reference/hooks/formats/predefined/#cssvariables)) supporting additional [Otter Extensions features](#enhancement). |
```

> | o3r/json/metadata | See [css/variables options](https://styledictionary.com/reference/hooks/formats/predefined/#cssvariables) *(applied to `defaultValue`)* | Renderer [CMS Style Metadata](https://github.com/AmadeusITGroup/otter/blob/main/docs/styling/THEME.md).                                                                                                 |

Reviewer: ```suggestion
| o3r/json/metadata | See [css/variables options](https://styledictionary.com/reference/hooks/formats/predefined/#cssvariables) *(applied to `defaultValue`)* | Render [CMS style metadata](https://github.com/AmadeusITGroup/otter/blob/main/docs/styling/THEME.md).                                                                                                 |
```

### Registration process

> The different hooks/modules can be register individually via the different [register...()](https://styledictionary.com/getting-started/using_the_npm_module/) functions from `StyleDictionary` or all at once with the `register` function as following :

Reviewer: ```suggestion
The different hooks/modules can be registered individually via the different [register...()](https://styledictionary.com/getting-started/using_the_npm_module/) functions from `StyleDictionary` or all at once with the `register` function as follows:
```

```typescript
import { register, baseConfig } from '@o3r/style-disctionary';

const sd = new StyleDictionary({
  ...baseConfig,
  // custom configs ...
});

register(sd); // Register all Otter modules
```

> [!IMPORTANT]
> To be able to use any of the listed modules/hooks, they have to be registered to the `StyleDefinition` instance.

> [!TIP]

> > The helper `baseConfig` will provide the minimal config for Otter extension support.

Reviewer: ```suggestion
> The helper `baseConfig` will provide the minimal configuration for Otter extension support.
```

### Helpers

To enhance and facilitate the configuration of the `StyleDictionary` instance, the `@o3r/style-dictionary` exposes a set of helpers.

#### getTargetFiles

> The function `getTargetFiles` is used to parameter the file generated based on Design Token (following the same logic of `$extensions`):

Reviewer: ```suggestion
The `getTargetFiles` function is used to parameterize the generated file based on Design Token (following the same logic as `$extensions`):
```
